### ISSUE TITLE
I can't train my custom model

### ISSUE THREAD

**Issue by FantasyJay** [ultralytics/yolov5#9206]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I tried the path several times and finally failed to complete the training using the absolute path.
and the error returned: Exception: val: Error loading data from /Users/jayliu/Code/ImageToSpeech/train_dataset/train/labels: val: No images found

![image](https://user-images.githubusercontent.com/42663790/187226474-aa035034-2a0c-4256-8ce1-e35ed8147472.png)

**Comment by FantasyJay** [comment]
That's **stop at: "train: New cache created"** and **miss the correct step of "train: Caching images"**
@Glenn Jocher @ultralytics
![image](https://user-images.githubusercontent.com/42663790/187228259-b1641559-390b-4fda-8d88-1674dfd1f371.png)

**Comment by glenn-jocher** [comment]
@FantasyJay 👋 Hello! Thanks for asking about YOLOv5 🚀 **dataset formatting**. To train correctly your data must be in YOLOv5 format. Please see our [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) tutorial for full documentation on dataset setup and all steps required to start training your first model. A few excerpts from the tutorial:

### 1.1 Create dataset.yaml

[COCO128](https://www.kaggle.com/datasets/ultralytics/coco128) is an example small tutorial dataset composed of the first 128 images in [COCO](http://cocodataset.org/#home) train2017. These same 128 images are used for both training and validation to verify our training pipeline is capable of overfitting. [data/coco128.yaml](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), shown below, is the dataset config file that defines 1) the dataset root directory `path` and relative paths to `train` / `val` / `test` image directories (or *.txt files with image paths) and 2) a class `names` dictionary:
```yaml
# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: ../datasets/coco128  # dataset root dir
train: images/train2017  # train images (relative to 'path') 128 images
val: images/train2017  # val images (relative to 'path') 128 images
test:  # test images (optional)

# Classes (80 COCO classes)
names:
  0: person
  1: bicycle
  2: car
  ...
  77: teddy bear
  78: hair drier
  79: toothbrush

```


### 1.2 Create Labels

After using a tool like [Roboflow Annotate](https://roboflow.com/annotate?ref=ultralytics) to label your images, export your labels to **YOLO format**, with one `*.txt` file per image (if no objects in image, no `*.txt` file is required). The `*.txt` file specifications are:

- One row per object
- Each row is `class x_center y_center width height` format.
- Box coordinates must be in **normalized xywh** format (from 0 - 1). If your boxes are in pixels, divide `x_center` and `width` by image width, and `y_center` and `height` by image height.
- Class numbers are zero-indexed (start from 0).

<p align="center"><img width="750" src="https://user-images.githubusercontent.com/26833433/91506361-c7965000-e886-11ea-8291-c72b98c25eec.jpg"></p>

The label file corresponding to the above image contains 2 persons (class `0`) and a tie (class `27`):

<p align="center"><img width="428" src="https://user-images.githubusercontent.com/26833433/112467037-d2568c00-8d66-11eb-8796-55402ac0d62f.png"></p>


### 1.3 Organize Directories

Organize your train and val images and labels according to the example below. YOLOv5 assumes  `/coco128` is inside a `/datasets` directory **next to** the `/yolov5` directory. **YOLOv5 locates labels automatically for each image** by replacing the last instance of `/images/` in each image path with `/labels/`. For example:
```bash
../datasets/coco128/images/im0.jpg  # image
../datasets/coco128/labels/im0.txt  # label
```

<p align="center"><img width="700" src="https://user-images.githubusercontent.com/26833433/134436012-65111ad1-9541-4853-81a6-f19a3468b75f.png"></p>
</details>

Good luck 🍀 and let us know if you have any other questions!

**Comment by FantasyJay** [comment]
> @FantasyJay 👋 Hello! Thanks for asking about YOLOv5 🚀 **dataset formatting**. To train correctly your data must be in YOLOv5 format. Please see our [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) tutorial for full documentation on dataset setup and all steps required to start training your first model. A few excerpts from the tutorial:
> 
> ### 1.1 Create dataset.yaml
> [COCO128](https://www.kaggle.com/datasets/ultralytics/coco128) is an example small tutorial dataset composed of the first 128 images in [COCO](http://cocodataset.org/#home) train2017. These same 128 images are used for both training and validation to verify our training pipeline is capable of overfitting. [data/coco128.yaml](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), shown below, is the dataset config file that defines 1) the dataset root directory `path` and relative paths to `train` / `val` / `test` image directories (or *.txt files with image paths) and 2) a class `names` dictionary:
> 
> ```yaml
> # Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
> path: ../datasets/coco128  # dataset root dir
> train: images/train2017  # train images (relative to 'path') 128 images
> val: images/train2017  # val images (relative to 'path') 128 images
> test:  # test images (optional)
> 
> # Classes (80 COCO classes)
> names:
>   0: person
>   1: bicycle
>   2: car
>   ...
>   77: teddy bear
>   78: hair drier
>   79: toothbrush
> ```
> 
> ### 1.2 Create Labels
> After using a tool like [Roboflow Annotate](https://roboflow.com/annotate?ref=ultralytics) to label your images, export your labels to **YOLO format**, with one `*.txt` file per image (if no objects in image, no `*.txt` file is required). The `*.txt` file specifications are:
> 
> * One row per object
> * Each row is `class x_center y_center width height` format.
> * Box coordinates must be in **normalized xywh** format (from 0 - 1). If your boxes are in pixels, divide `x_center` and `width` by image width, and `y_center` and `height` by image height.
> * Class numbers are zero-indexed (start from 0).
> 
> <img alt="" width="750" src="https://user-images.githubusercontent.com/26833433/91506361-c7965000-e886-11ea-8291-c72b98c25eec.jpg">
> 
> The label file corresponding to the above image contains 2 persons (class `0`) and a tie (class `27`):
> 
> <img alt="" width="428" src="https://user-images.githubusercontent.com/26833433/112467037-d2568c00-8d66-11eb-8796-55402ac0d62f.png">
> 
> ### 1.3 Organize Directories
> Organize your train and val images and labels according to the example below. YOLOv5 assumes `/coco128` is inside a `/datasets` directory **next to** the `/yolov5` directory. **YOLOv5 locates labels automatically for each image** by replacing the last instance of `/images/` in each image path with `/labels/`. For example:
> 
> ```shell
> ../datasets/coco128/images/im0.jpg  # image
> ../datasets/coco128/labels/im0.txt  # label
> ```
> 
> <img alt="" width="700" src="https://user-images.githubusercontent.com/26833433/134436012-65111ad1-9541-4853-81a6-f19a3468b75f.png">
> 
> Good luck 🍀 and let us know if you have any other questions!



I already set my dataset format to yolov5 with Roboflow. And still not work.

**Comment by barney2074** [comment]
post a screenshot of your yaml, data folder structure & contents of images/val 
From the error, it seems your images/val folder does not contain valid images

**Comment by glenn-jocher** [comment]
@barney2074 i appreciate your patience. To resolve this issue, let's first ensure that your data folder structure and contents adhere to the YOLOv5 format. 

Could you please post a screenshot of your `yaml` file, data folder structure, and the contents of `images/val`? It seems that your `images/val` folder may not contain valid images. This will help in identifying any potential issues with your dataset setup.

Thank you for your cooperation!
